packages/editor-ui/src/views/NodeView.vue

```diff
@@ -4789,11 +4789,13 @@ export default defineComponent({
 		},
 		async loadCredentials(): Promise<void> {
 			const workflow = this.workflowsStore.getWorkflowById(this.currentWorkflow);
-			const projectId =
-				workflow?.homeProject?.type === ProjectTypes.Personal
+			const projectId = !workflow
+				? (this.$route.query?.projectId as string | undefined) ??
+					this.projectsStore.personalProject?.id
+				: workflow?.homeProject?.type === ProjectTypes.Personal
 					? this.projectsStore.personalProject?.id
-					: workflow?.homeProject?.id;
-			await this.credentialsStore.fetchAllCredentials(projectId);
+					: workflow?.homeProject?.id ?? this.projectsStore.currentProjectId;
```

Comment:
This get's a bit hard to read. What do you think about this? It's also a bit more type safe because it does not cast, but type guard instead:

```ts
			let projectId: string | undefined;
			if (workflow) {
				projectId =
					workflow.homeProject?.type === ProjectTypes.Personal
						? this.projectsStore.personalProject?.id
						: workflow?.homeProject?.id ?? this.projectsStore.currentProjectId;
			} else {
				const queryParam =
					typeof this.$route.query?.projectId === 'string'
						? this.$route.query?.projectId
						: undefined;
				projectId = queryParam ?? this.projectsStore.personalProject?.id;
			}
```

Reply:
Tests pass locally.

Reply:
Sure